### PR TITLE
Verify that firewall is still there when reconciling infrastructure.

### LINKED ISSUE #15: firewall infrastructure does not get reconciled
How to reproduce:

- create cluster
- cluster has two workers and one firewall:

```
metalctl machine list --project ed53d0dc-fd1c-41f9-a534-bf84c61c98c9
  ID                                         LAST EVENT   WHEN  AGE      HOSTNAME                        PROJECT                               SIZE           IMAGE         PARTITION
  fafd0c00-7090-11e9-8000-efbeaddeefbe       Phoned Home  2s    23m 9s   shoot--p7l8m...-firewall-fcbab  ed53d0dc-fd1c-41f9-a534-bf84c61c98c9  c1-xlarge-x86  Firewall 1    nbg-w8101
  00000000-beef-beef-0011-efbeaddeefbe       Phoned Home  3s    18m 52s  shoot--p7l8m...dfb8b55dc-f9zxg  ed53d0dc-fd1c-41f9-a534-bf84c61c98c9  c1-xlarge-x86  Ubuntu 19.04  nbg-w8101
  00000000-beef-beef-0001-efbeaddeefbe       Phoned Home  1s    6m 36s   shoot--p7l8m...dfb8b55dc-nfvv9  ed53d0dc-fd1c-41f9-a534-bf84c61c98c9  c1-xlarge-x86  Ubuntu 19.04  nbg-w8101

```
- destroy firewall
`metalctl machine rm fafd0c00-7090-11e9-8000-efbeaddeefbe
`
- reconcile cluster
`cloudctl cluster reconcile c1605f39-1668-11ea-853c-42ced4c4a306`

=> reconcile freezes at 86%:

```
  UID                                   TENANT  PROJECT                               NAME      VERSION  PARTITION  OPERATION   PROGRESS         API   CONTROL  NODES  SYSTEM  SIZE  AGE
  c1605f39-1668-11ea-853c-42ced4c4a306  fits    ed53d0dc-fd1c-41f9-a534-bf84c61c98c9  mwen4895  1.14.3   nbg-w8101  Processing  86% [Reconcile]  True  True     False  False   2/2   54m 7s

```
project still without firewall:
```
  ID                                         LAST EVENT   WHEN  AGE      HOSTNAME                        PROJECT                               SIZE           IMAGE         PARTITION
  00000000-beef-beef-0011-efbeaddeefbe       Phoned Home  15s   49m 6s   shoot--p7l8m...dfb8b55dc-f9zxg  ed53d0dc-fd1c-41f9-a534-bf84c61c98c9  c1-xlarge-x86  Ubuntu 19.04  nbg-w8101
  00000000-beef-beef-0001-efbeaddeefbe       Phoned Home  17s   36m 50s  shoot--p7l8m...dfb8b55dc-nfvv9  ed53d0dc-fd1c-41f9-a534-bf84c61c98c9  c1-xlarge-x86  Ubuntu 19.04  nbg-w8101

```

### PR DESCRIPTION
**Pull request by Gerrit91**
Closes #15.

I added a validation section if the firewall still exists and belongs to this cluster in the infrastructure reconciler. 

This change would cause re-creation of cluster firewalls because I added a machine tag used for identifying if a firewall belongs to a cluster or not.

### CODE REVIEW

File: pkg/controller/infrastructure/actuator_reconcile.go [hunk 1]
```diff
@@ -34,18 +36,56 @@ func (a *actuator) reconcile(ctx context.Context, infrastructure *extensionsv1al
...
+	firewallStatus := infrastructureStatus.Firewall
+
+	if firewallStatus.Succeeded {
+		// verify that the firewall is still there and correctly reconciled
+		machineID := decodeMachineID(firewallStatus.MachineID)
+
+		resp, err := mclient.FirewallFind(&metalgo.FirewallFindRequest{
+			MachineFindRequest: metalgo.MachineFindRequest{
+				ID:                &machineID,
+				AllocationProject: &infrastructureConfig.ProjectID,
```
Comment (majst01) [inline pkg/controller/infrastructure/actuator_reconcile.go]: we can add clusterid tag here as well to narrow down the search term. Just thought.

**Comment by majst01** [comment]
In the delete Firewall actuator we should check for machineID and ProjectID and probable ClusterTag as well

File: pkg/controller/infrastructure/actuator_reconcile.go (hunk 1 above)
Comment (Gerrit91) [reply pkg/controller/infrastructure/actuator_reconcile.go]: Good idea, this also reduces the amount of changes.

**Comment by Gerrit91** [comment]
Added the find request to the delete reconcile algorithm as well.

**Comment by Gerrit91** [comment]
I would say... let's just try it out and fix if there are issues.

**Pull request merged**